pytorch_lightning/core/lightning.py

```diff
@@ -1598,7 +1598,7 @@ def __init__(self, hparams: dict):
 
         # override the hparam keys that were passed in
         if hparam_overrides is not None:
-            update_hparams(hparams, hparam_overrides)
+            update_hparams(checkpoint['hparams'], hparam_overrides)
```

Comment:
Your fix would theoretically work, but I prefer to just move the update function before 

```
# overwrite hparams by the given file
checkpoint['hparams'] = hparams
```

which would have the same effect but porbably is a bit cleaner

Reply:
Yes that makes sense. I'll update the PR.

Reply:
mind add test for it?

Reply:
@justusschock 
Actually, the ability to overwrite `hparams` that were passed through a .csv/.yaml file could be useful. If I move this update before we load the `hparams` from file we loose that. 

You know better, so I'll still update the PR if this is not the desired functionality.

Reply:
I'll look into adding a test @Borda.

Reply:
you shouldn't move this before we load them but before you reinject them to the checkpoint :)